docs/language-server/protocol-project-manager.md

```diff
@@ -82,6 +109,12 @@ specified project.
 ```typescript
 interface ProjectOpenRequest {
   projectId: UUID;
+
+  /** Specifies whether to install missing components.
+   *
+   * If not provided, defaults to false.
+   */
+  installMissingComponents?: bool;
```

Comment:
One more thing that we may need to discuss.

Apart from a component just being missing, the version that is required may be missing *and* marked as broken. The launcher does not allow installing `broken` versions by default, unless forced and I guess the IDE should also display some kind of warning about the version being broken and suggesting to upgrading to a more recent version.

My suggestion is to add an analogous `forceInstallBrokenComponents?: bool` that will default to `false` and will normally be set as such. Then if the missing engine is marked as broken the operation could fail with `MissingComponentIsBroken` error and that can be recovered by the IDE in a similar way missing components are.

Analogous parameter may or may not be needed for `project/create`. Alternatively, create could proceed with broken versions, as before selecting the version to create, you need to find that this version even exists and `engine/list-available` can include that information. Or maybe the broken releases should be completely ignored for purposes of creating new projects?

Reply:
> My suggestion is to add an analogous forceInstallBrokenComponents?: bool that will default to false and will normally be set as such. Then if the missing engine is marked as broken the operation could fail with MissingComponentIsBroken error and that can be recovered by the IDE in a similar way missing components are.

Ok, however one enum with three variants is better than two bools. Or with two variants, as this is optional field, as you please

Concerning `project/create`: I would keep the same option as in `project/open`, for simplicity. IDE may just don't use all variants, or e.g. guard creating project with broken versions with three irritating warning dialogs.